src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -548,31 +562,73 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
+                if (configSection.TryGetValue("consoleLevel", out value))
+                {
+                    TraceLevel consoleLevel;
+                    if (Enum.TryParse<TraceLevel>((string)value, true, out consoleLevel))
+                    {
+                        hostConfig.Tracing.ConsoleLevel = consoleLevel;
+                    }
+                }
+
+                if (configSection.TryGetValue("fileLoggingEnabled", out value))
                 {
-                    hostConfig.Tracing.ConsoleLevel = consoleLevel;
+                    scriptConfig.FileLoggingEnabled = (bool)value;
                 }
             }
 
+            // Set up a host level TraceMonitor that will receive notificaition
+            // of ALL errors that occur. This allows us to inspect/log errors.
+            var traceMonitor = new TraceMonitor()
+                .Filter(p => { return true; })
+                .Subscribe(HandleHostError);
+            hostConfig.Tracing.Tracers.Add(traceMonitor);
+
             hostConfig.UseTimers();
             hostConfig.UseCore();
         }
 
-        private void AddFunctionError(string functionName, string error)
+        private void HandleHostError(Microsoft.Azure.WebJobs.Extensions.TraceFilter traceFilter)
         {
-            Collection<string> errors = null;
-            if (!FunctionErrors.TryGetValue(functionName, out errors))
+            foreach (TraceEvent traceEvent in traceFilter.Events)
             {
-                errors = new Collection<string>();
-                FunctionErrors[functionName] = errors;
+                if (traceEvent.Exception is FunctionInvocationException)
+                {
+                    // For all function invocation events, we notify the invoker so it can
+                    // log the error as needed to its function specific logs.
+                    FunctionInvocationException invocationException = traceEvent.Exception as FunctionInvocationException;
+                    NotifyInvoker(invocationException.MethodName, invocationException);
+                }
+                else if (traceEvent.Exception is FunctionIndexingException)
+                {
+                    // For all startup time indexing errors, we accumulate them per function
+                    FunctionIndexingException indexingException = traceEvent.Exception as FunctionIndexingException;
+                    string formattedError = Utility.FlattenException(indexingException);
+                    AddFunctionError(indexingException.MethodName, formattedError);
+
+                    // Also notify the invoker so the error can also be written to the function
+                    // log file
+                    NotifyInvoker(indexingException.MethodName, indexingException);
+
+                    // Mark the error as handled so indexing will continue
+                    indexingException.Handled = true;
```

Comment:
This is the key - it tells the core SDK to continue indexing the rest of the methods (i.e. don't blow up)